rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -44,38 +44,36 @@ void Recorder::record(const RecordOptions & record_options)
 
   std::future<void> discovery_future;
   if (!record_options.is_discovery_disabled) {
-    discovery_future = launch_topics_discovery(
+    auto launch_discovery = std::bind(
```

Comment:
I prefer the lambda syntax over bind.

Reply:
i call the function directly because I don't see the point in calling a function which just wraps a lambda.